internal/provider/providers/duckdns/provider.go

```diff
@@ -146,7 +146,11 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 	case s[0:minChars] == "KO":
 		return netip.Addr{}, fmt.Errorf("%w", errors.ErrAuth)
 	case s[0:minChars] == "OK":
-		ips := ipextract.IPv4(s)
+		if ip.Is6() {
+			ips := ipextract.IPv6(s)
+		} else {
+		        ips := ipextract.IPv4(s)
+		}
```

Comment:
```suggestion
                var ips []netip.Addr
		if ip.Is6() {
			ips = ipextract.IPv6(s)
		} else {
		        ips = ipextract.IPv4(s)
		}
```
found this in the no-ip provider code.

Reply:
Yes that's correct 😉